# Practice assignment for Lab1

## This assignment is not mandatory and not graded. It is rather an opportunity to start practicing basic concepts in Python which you will also use in the assignents. 

### In this practice assignment you will read the iris dataset, explore it, do some basic data pre-processing and finally create some plots to visualize your variables!


# Part 1

1. Import pandas

In [2]:
import pandas as pd

2. Read the iris dataset which is under the datasets folder using the relative path. 

In [3]:
data = pd.read_csv("../datasets/iris.data",header=None)

In [4]:
data

0    1    2    3               4
0    5.1  3.5  1.4  0.2     Iris-setosa
1    4.9  3.0  1.4  0.2     Iris-setosa
2    4.7  3.2  1.3  0.2     Iris-setosa
3    4.6  3.1  1.5  0.2     Iris-setosa
4    5.0  3.6  1.4  0.2     Iris-setosa
..   ...  ...  ...  ...             ...
145  6.7  3.0  5.2  2.3  Iris-virginica
146  6.3  2.5  5.0  1.9  Iris-virginica
147  6.5  3.0  5.2  2.0  Iris-virginica
148  6.2  3.4  5.4  2.3  Iris-virginica
149  5.9  3.0  5.1  1.8  Iris-virginica

[150 rows x 5 columns]

7. Attribute Information:


   1. sepal length in cm
   2. sepal width in cm
   3. petal length in cm
   4. petal width in cm
   5. class: 


      1.  Iris Setosa
      2.  Iris Versicolour
      3.  Iris Virginica


3. Print the 5 first rows of the table 

4.  Create a list named feature_names that contains the above names ("sepal length" , "sepal width", "petal length", "petal width", "class") and assign them as feature names

5. Use the .info() method to find out how many entries there are and their type

6. Print the average and max petal width

In [ ]:
"""
Use the max() and mean() built-in Python functions to find the the max and average petal width. 

Print the results using the print() function. 

"""




7. Find the number of different values  of the class label 

In [ ]:
"""
Retrieve the column that contains the class label and use the function value_counts()

"""



8. Get a subset of the original dataframe where the class label equals to Iris-versicolor

9. Get a subset of the dataframe where the class label is all the other labels but not Iris-versicolor


In [ ]:
"""

Hint: the not equal operator in Python is !=

Essentially copy and paste what you did in question 8 and change the operator

"""



10. What is the highest sepal length of each class type?

In [ ]:
"""
Use the groupby() function to group by class and then use a summarization function on sepal length

"""



11. Extract only the first two columns of the dataframe by a) using the names of the columns and b) using iloc


In [ ]:
#or:


12. Extract the the first two rows:


13. Create a series with as many (random) string values as the rows of the dataframe and attach this new series as a new column of the iris dataframe


In [ ]:
"""
Hint:
Create a list with your 5 favorite colors, multiple it by 30, convert it to series and attach it to the df along the column axis using the concat() method

"""




# Part 2


## In this part you will use matplotlib and seaborn to plot

1. Import matplpotlib

2. Plot the petal length for all the 150 flowers

In [ ]:
"""

1. Create your empty canvas by fig = plt.figure()
2. Use te add_axes() 
3. Use the plot() function in axes module to plot the flower samples versus the petal length of each as lines
4. Add labels on the y and  x axes
5. Set a title on the figure


"""

fig = plt.figure()


3. Create a scatter plot of petal length against the petal width and add a grid



In [ ]:
"""
This time you can call plot directly from the dataframe (do not define a figure object). 
As parameters you need to specify the kind of plot you want (scatter), then the column name to be used as horizontal coordinates (sepal length) and the column name to be used as vertical coordinated (sepal width)
add a grid to the figure!
"""
data.plot(....)


4. Create ONE scatter plot of sepal length against the petal length for all the Iris species(class labels). 

In [ ]:
"""
1. Create a list of the colors that you want to assign to each class label. 
2. Create a list of the unique values of the class label
3. Loop through the list of unique class labels of step 2 using enumerate()
4. Create a sub-dataframe inside the loop that contains all the rows of the specific class label that you are looping through
5. Create a scatterplot for the sub-dataframe with horizontal coordinates the sepal length and vertical coordinates the sepal width
6. Assign a title and labels to each axis. 
7. Use legend to print on the plot the the colors that correspond on the class labels. 

"""



5. Use seaborn and make a pairplot of all the numerical values of the iris dataset and pass as an attribute hue="class"

In [ ]:
"""
Do not forget to import seaborn first!

"""


# End of practice assignment 1